server/test/unit/server/async/test_app.py

```diff
@@ -16,17 +24,87 @@ class InitializeWorkerTestCase(unittest.TestCase):
+    def test_initialize_worker_resource_manager(self,
+                                                mock_get_resource_manager_lock,
+                                                _delete_worker, initialize,
+                                                create_worker_working_directory,
+                                                delete_worker_working_directory):
+        """
+        Assert that initialize_worker() calls Pulp's initialization code and the appropriate worker
+        monitoring code for a resource mananger worker.
         """
-        sender = mock.MagicMock()
-        # The args aren't used and don't matter, so we'll just pass some mocks
+        sender = RESOURCE_MANAGER_WORKER_NAME + '@' + platform.node()
+        # The instance argument isn't used and don't matter, so we'll just pass a mock
         app.initialize_worker(sender, mock.MagicMock())
 
         initialize.assert_called_once_with()
         _delete_worker.assert_called_once_with(sender, normal_shutdown=True)
         create_worker_working_directory.assert_called_once_with(sender)
         delete_worker_working_directory.assert_called_once_with(sender)
+        mock_get_resource_manager_lock.assert_called_once_with(sender)
+
+    @mock.patch('pulp.server.async.app.datetime')
+    @mock.patch('pulp.server.async.app.Worker')
+    @mock.patch('pulp.server.async.app.ResourceManagerLock')
+    def test_get_resource_manager_lock(self, mock_rm_lock, mock_worker, mock_datetime):
+        """
+        Assert that get_resource_manager_lock() attempts to save a lock to the database with the
+        correct name, and that it creates a worker entry with that name and the correct timestamp.
+        """
+        sender = RESOURCE_MANAGER_WORKER_NAME + '@' + platform.node()
+        app.get_resource_manager_lock(sender)
+
+        mock_rm_lock.assert_called_once_with(name=sender)
+        mock_rm_lock.objects().save().assert_called_once()
+        mock_worker.objects(name=sender).update_one.\
+            assert_called_once_with(set__last_heartbeat=mock_datetime.utcnow(), upsert=True)
+
+    @mock.patch('pulp.server.async.app.sys')
+    @mock.patch('pulp.server.async.app.tasks._delete_worker')
+    def test_custom_sigterm_handler(self, _delete_worker, mock_sys):
+        """
+        Assert that the custom_sigterm_handler context manager properly sets and restores the
+        SIGTERM signal handler upon entry and exit, and that it calls the delete_worker cleanup
+        routine with the correct worker name and then exits.
+        """
+        handler = signal.getsignal(signal.SIGTERM)
+        self.assertEquals(handler, signal.SIG_DFL)
+
+        name = RESOURCE_MANAGER_WORKER_NAME + '@' + platform.node()
+
+        with app.custom_sigterm_handler(name):
+            handler = signal.getsignal(signal.SIGTERM)
+            self.assertNotEquals(handler, signal.SIG_DFL)
+
+            handler(None, None)
```

Comment:
This would be good as a separate test. A goal for a unit test should be that if it fails, the developer knows exactly what broke. This test is currently testing two different things. 1) did the context manager swap the signal handler in and out, and 2) is the custom signal handler doing its job.

Even though the setup for both tests will be similar or identical, and it seems convenient to assert both things in the same test, the test suite will be more useful if we separate them.